File: relaize/backend/app/schemas/catalog.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class ModelInfo(BaseModel):
    id: str
    name: str
    kind: str
    description: str
    repo: Optional[str] = None
    homepage: Optional[str] = None
    tags: list[str] = Field(default_factory=list)
    default_device: str = "cuda"
    weight_hint: Optional[str] = None
    supports_prompt: bool = False
    supports_mask: bool = False


class PipelineStageInfo(BaseModel):
    id: str
    name: str
    model_id: str
    description: str
    optional: bool = False
    defaults: dict = Field(default_factory=dict)


class PipelineInfo(BaseModel):
    id: str
    name: str
    description: str
    tags: list[str] = Field(default_factory=list)
    stages: list[PipelineStageInfo] = Field(default_factory=list)
    recommended_presets: list[str] = Field(default_factory=list)
    supports_prompt: bool = False
    supports_mask: bool = False


class CatalogResponse(BaseModel):
    models: list[ModelInfo]
    pipelines: list[PipelineInfo]

File: relaize/frontend/src/types/catalog.ts
export type ModelCatalogEntry = {
  id: string;
  name: string;
  kind: string;
  description: string;
  repo?: string | null;
  homepage?: string | null;
  tags?: string[];
  default_device?: string;
  weight_hint?: string | null;
  supports_prompt?: boolean;
  supports_mask?: boolean;
};

export type PipelineStageInfo = {
  id: string;
  name: string;
  model_id: string;
  description: string;
  optional?: boolean;
  defaults?: Record<string, unknown>;
};

export type PipelineCatalogEntry = {
  id: string;
  name: string;
  description: string;
  tags?: string[];
  stages: PipelineStageInfo[];
  recommended_presets?: string[];
  supports_prompt?: boolean;
  supports_mask?: boolean;
};
